pkg/canary/reader/reader.go

```diff
@@ -442,6 +459,35 @@ func (r *Reader) closeAndReconnect() {
 	}
 }
 
+// httpClient uses the config in Reader to return a http client.
+// http.DefaultClient will be returned in the case that the connection to Loki is http or TLS without client certs.
+// For the mTLS case, return a http.Client configured to use the client side certificates.
+func (r *Reader) httpClient() (*http.Client, error) {
+	if r.clientTLSConfig == nil {
+		return http.DefaultClient, nil
+	}
+	rt, err := config.NewTLSRoundTripper(r.clientTLSConfig, r.caFile, func(tls *tls.Config) (http.RoundTripper, error) {
+		return &http.Transport{TLSClientConfig: tls}, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &http.Client{
+		Transport: rt,
+	}, nil
+}
+
+// webSocketDialer creates a dialer for the web socket connection to Loki
+// websocket.DefaultDialer will be returned in the case that the connection to Loki is http or TLS without client certs.
+// For the mTLS case, return a websocket.Dialer configured to use client side certificates.
+func (r *Reader) webSocketDialer() *websocket.Dialer {
+	dialer := websocket.DefaultDialer
+	if r.clientTLSConfig != nil {
+		dialer.TLSClientConfig = r.clientTLSConfig
```

Comment:
We probably shouldn't be mutating `websocket.DefaultDialer` in this library package.

Instead, let's create a new one like:
```
var dialer = &Dialer{
	Proxy:            http.ProxyFromEnvironment,
	HandshakeTimeout: 45 * time.Second,
	TLSClientConfig = r.clientTLSConfig,
}
```

Reply:
My understanding is that `dialer` is a copy of `websocket.DefaultDialer` and there is no mutation happening outside this package. However, I like your way better as it keeps me from having to do local mutation.